### ISSUE TITLE
Delay between wakeword and recording start

### ISSUE THREAD

**Issue by david-morris** [MycroftAI/mycroft-core#2882]
**Describe the bug**
When I have something playing on Kodi on the same host as my Mycroft instance, the wakeword will trigger and then 30 seconds to 2 minutes will pass before Mycroft actually reduces the volume of playback and listens.

**To Reproduce**
1. Start kodi playback on the same host by any means
2. use the wake word

**Expected behavior**
Volume lowers and recording begins within three seconds of the wakeword being recognized.

**Log files**
`voice.log`:
```
2021-04-16 10:19:56.387 | INFO     | 3764518 | __main__:handle_utterance:76 | Utterance: ['play german news from favorites']
2021-04-16 10:32:57.957 | INFO     | 3764518 | mycroft.session:get:72 | New Session Start: 0ac028f4-54d6-4908a40f-4820f3bc4c7
2021-04-16 10:32:57.958 | INFO     | 3764518 | __main__:handle_wakeword:71 | Wakeword Detected: computer
Failed to drain stream: Timeout
2021-04-16 10:33:28.023 | INFO     | 3764518 | __main__:handle_record_begin:41 | Begin Recording...
2021-04-16 10:33:30.791 | INFO     | 3764518 | __main__:handle_record_end:49 | End Recording...
2021-04-16 10:33:32.095 | INFO     | 3764518 | __main__:handle_utterance:76 | Utterance: ['stop kodi']
2021-04-16 10:45:27.380 | INFO     | 3764518 | mycroft.session:get:72 | New Session Start: c3035158-38c7-49c8ac09-5acabf8c2b79
2021-04-16 10:45:27.382 | INFO     | 3764518 | __main__:handle_wakeword:71 | Wakeword Detected: computer
2021-04-16 10:45:27.763 | INFO     | 3764518 | __main__:handle_record_begin:41 | Begin Recording...
2021-04-16 10:45:37.752 | INFO     | 3764518 | __main__:handle_record_end:49 | End Recording...
2021-04-16 10:45:44.333 | INFO     | 3764518 | __main__:handle_utterance:76 | Utterance: ['move kodi cursor down']
```

**Environment (please complete the following information):**
 - Device type: halftop (x86_64) running headless except for a wayland kiosk instance of kodi
 - OS: Arch
 - Mycroft-core version: 20.8.1
 - Other versions: pulseaudio 14.2, python 3.9.2 (yes inside the .venv).

**Additional context**

I use mycroft on a system set up as a kodi/general server arch box. Pulseaudio is started by kodi when the kodi-standalone service starts kodi-wayland (under a different user).  I have tested playing a video back on another computer in the same room, and that makes triggering the wake word harder, but once triggered, Mycroft does not delay before recording, nor does the default speech recognition have any trouble picking out background noise.

`mycroft.conf`:
```
{
  "max_allowed_core_version": 20.8,
  "skills": {
    "blacklisted_skills": [
      "mycroft-npr-news",
      "mycroft-stock"
    ]
  },
  "listener": {
    "wake_word": "computer",
    "mute_during_output": false
  },
  "hotwords": {
    "hey quinoa": {
      "module": "pocketsphinx",
      "phonemes": "HH EY . K IY N W AO.",
      "threshold": 1e-05
    },
    "computer": {
      "module": "porcupine_wakeword_plug",
      "keyword_file_path": "~/.mycroft/computer_linux.ppn",
      "sensitivity": 0.7
    }
  }
}
```

Something might be weird with my multiprocessing setup, although I haven't chosen to touch it, this happens on startup (though padatious intents still work fine):

```
2021-04-15 09:33:09.006 | INFO     | 3764511 | mycroft.skills.intent_services.padatious_service:train:81 | Training... (single_thread=False)
Exception in thread Thread-7:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.9/threading.py", line 892, in run
    self._target(*self._args, **self._kwargs)
  File "/home/dmorris/src/mycroft-core/.venv/lib/python3.9/site-packages/padatious/training_manager.py", line 98, in train
    pool = mp.Pool()
  File "/usr/lib/python3.9/multiprocessing/context.py", line 119, in Pool
    return Pool(processes, initializer, initargs, maxtasksperchild,
  File "/usr/lib/python3.9/multiprocessing/pool.py", line 191, in __init__
    self._setup_queues()
  File "/usr/lib/python3.9/multiprocessing/pool.py", line 343, in _setup_queues
    self._inqueue = self._ctx.SimpleQueue()
  File "/usr/lib/python3.9/multiprocessing/context.py", line 112, in SimpleQueue
    from .queues import SimpleQueue
ImportError: cannot import name 'SimpleQueue' from partially initialized module 'multiprocessing.queues' (most likely due to a circular import) (/usr/lib/python3.9/multiprocessing/queues.py)
```

**Comment by JarbasAl** [comment]
your multiprocessing errors seem related to #2799

**Comment by david-morris** [comment]
I just noticed that sometimes dialog is buffered while I have media playing, and all gets spoken when the media stops.

**Comment by david-morris** [comment]
I've been adding some debug statements to `mycroft/client/speech/mic.py` MutableStream to try and figure out what's going on here, since this interferes with my use case of mycroft quite heavily.  I've noticed some things.

- In normal functionality, the stream is always enabled and `unmute()` is called when the wakeword is detected.
- When this bug occurs, `mute()` is called instead of `unmute()` when the wakeword is detected.  Some time later, `unmute()` is called and recording begins.
- `is_muted()` never gets called in any of this.
- This bug occurs whether media is started from mycroft skills or other kodi remotes.
- This bug can happen shortly after media is closed.

**Comment by david-morris** [comment]
@JarbasAl , I've just tried checking out forslund's bugfix/mp-workaround branch. That stops the error during padatious loading, but this bug still occurs.

**Comment by david-morris** [comment]
OK, I've tracked this down. It appears that this is an issue where when there is a stream from a different user playing, attempts to play things via Pulseaudio will hang, so play_wav hangs.  If I can confirm this, I'll close this issue.

If you're a kodi-standalone user, I'll post the solution in the closing comment.

**Comment by david-morris** [comment]
I was able to confirm that this issue was due to my pulseaudio settings. I followed the instructions [here on the archwiki](https://wiki.archlinux.org/index.php/PulseAudio/Examples#Allowing_multiple_users_to_use_PulseAudio_at_the_same_time) for using UNIX sockets to erase pulseaudio's knowledge of which user was opening an audio stream, except I changed the files in `/etc/pulse` against that article's recommendations because of quirks of the kodi-standalone packages.

Now I am able to play sounds from different users at the same time, and mycroft does not hang waiting for pulseaudio.
